src/jit/gentree.cpp

```diff
@@ -6746,42 +7009,36 @@ void GenTreeIntCon::FixupInitBlkValue(var_types asgType)
 
 // Initializes a BlkOp GenTree
 // Preconditions:
-//     - Result is a GenTreeBlkOp that is newly constructed by gtNewCpObjNode or gtNewBlkOpNode
+//     - Result is an assignment that is newly constructed
 //
 // Parameters:
-//     - result is a GenTreeBlkOp node that is the node to be initialized.
-//     - oper must be either GT_INITBLK or GT_COPYBLK
+//     - result is an assignment node that is to be initialized.
 //     - dst is the target (destination) we want to either initialize or copy to
```

Comment:
Indeed, if it's not TYP_STRUCT, it should do an early return. I'll make that change as well.

---
In reply to: [77396814](https://github.com/dotnet/coreclr/pull/7027#discussion_r77396814) [](ancestors = 77396814,77389617)